<launch>

  <!-- This node updates the 'position_yaw_init' parameter of a target MSF sensor node using yaw information received from an external topic. Various forms of yaw topic inputs are available (see below).

  To check if the node is working correctly, open the target MSF node parameters in a Dynamic Reconfigure client (using rqt). You should see the 'position_yaw_init' parameter changing to correspond to the input heading topic. The update frequency is 0.5Hz by default; this can be changed by controlling the rate at which this node is running. -->
  <node pkg="msf_updates" type="init_yaw_from_compass.py" name="init_yaw" clear_params="true" output="screen">

    <!-- Inputs -->
    <!-- Specify name of node running MSF -->
    <param name="msf_sensor_node" type="string" value="msf_gps_pose_estimator/position_sensor" />

    <!-- Specify topic to obtain heading from (use one) -->
    <remap from="imu" to="imu" />
    <!-- remap from="heading_radians" to="heading" / -->
    <!-- remap from="orientation_quaternion" to="fcu/imu/orientation" / -->
    <!-- remap from="orientation_degrees" to="orientation_euler_estimated_odometry_degrees" / -->

  </node>

</launch>
